src/eva/data/mon_data_space.py

```diff
@@ -0,0 +1,302 @@
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+import os
+import numpy as np
+from xarray import Dataset, concat
+
+from scipy.io import FortranFile
+
+from eva.eva_base import EvaBase
+from eva.utilities.config import get
+from eva.utilities.utils import parse_channel_list
+
+# --------------------------------------------------------------------------------------------------
+
+class MonDataSpace(EvaBase):
+
+    # ----------------------------------------------------------------------------------------------
+
+    def execute(self, data_collections, timing):
+
+        # Get name of the control file to parse
+        # -------------------------------------
+        ctlfile = self.config.get('ctlfile')
+        mystr=""
+        for x in ctlfile:
+            mystr += ' ' + x
+        self.logger.info('ctlfile: ' + mystr)
+
+        channo, nchans, nregion, satellite, sensor = self.get_ctl_stats(ctlfile[0])
+        self.logger.info('channo  = ' + str(channo))
+        self.logger.info('nchans  = ' + str(nchans))
+        self.logger.info('nregion = ' + str(nregion))
+        self.logger.info('satellite = ' + str(satellite))
+        self.logger.info('sensor = ' + str(sensor))
+
+        # Loop over the datasets
+        # ----------------------
+        for dataset in self.config.get('datasets'):
+
+            # Set the collection name
+            # -----------------------
+            collection_name = get(dataset, self.logger, 'name')
+
+            # Get the groups to be read
+            # -------------------------
+            groups = get(dataset, self.logger, 'groups')
+
+            # Get requested channels, convert to list
+            # ---------------------------------------
+            channels_str_or_list = get(dataset, self.logger, 'channels')
+            self.logger.info(' channels_str_or_list = ' + str(channels_str_or_list))
+
+            #  See regions below for explanation of str() conversion here.
+
+            # yaml file with no specified channels = use all channels
+            drop_channels = False
+            requested_channels = []
+
+            if channels_str_or_list is not None: 
+                self.logger.info(' inside not None')
+
+                if len(str(channels_str_or_list)) > 0:
+                    self.logger.info(' inside len check ')
+                    requested_channels = parse_channel_list(str(channels_str_or_list), self.logger)
+                    drop_channels = True
+
+            # Get requested regions, convert to list
+            # --------------------------------------
+            regions_str_or_list = get(dataset, self.logger, 'regions', [])
+            self.logger.info(' regions_str_or_list = ' + str(regions_str_or_list))
+
+            #  Note:  use of regions_str_or_list is lifted directly from gsi_obs_space.py but it 
```

Comment:
I think this is a bug.  I've documented the range of returned types in the code comments.  I note that if you try to run the testGsiObsSpaceAmsuaMetop-A.yaml with a single requested channel instead of 2 it fails, for the reason I've outlined.

Reply:
I just retested this and I'm convinced it is a bug.  If you edit `eva/src/eva/tests/config/testGsiObsSpaceAmsuaMetop-A.yaml` and change `channels: &channels 3,8` to `channels: &channels 3` this is the result:

`INFO EvaSetup: Starting Eva`
`INFO GsiObsSpace:   Initializing eva GsiObsSpace object`
`INFO EvaSetup: Running execute for GsiObsSpace`
`ABORT GsiObsSpace: In parse_channel_list the input is neither a list of integers or a string that can be parsed into a list of integers.`
`ABORT`

It's choking on 3 because it is not typed list of integers nor a string; it's a single integer.

Reply:
ah, yeah that is something we need to fix or else do `[3]` for in the meantime.

Reply:
I'll open a new issue for it then.

Reply:
Issue #71 was opened to handle this and PR #72 has been submitted for the fix.